airbyte/_util/api_util.py

```
@@ -1194,3 +1194,41 @@ def get_connector_builder_project_for_definition_id(
         client_secret=client_secret,
     )
     return json_result.get("builderProjectId")
+
+
+def get_definition_id_for_connector_builder_project(
+    *,
+    workspace_id: str,
+    builder_project_id: str,
+    api_root: str,
+    client_id: SecretString,
+    client_secret: SecretString,
+) -> str | None:
```

We should raise an error if not found. No need to allow null output here.